tests/TranslationExtractorTest.php

```diff
@@ -0,0 +1,106 @@
+
+        $extractor = new TranslationExtractor($notExistsPath);
+        $extractor->extract();
+    }
+
+    public function testExtractorWithOnlyCorrectData(): void
+    {
+        $path = __DIR__ . DIRECTORY_SEPARATOR . 'extractorExamples' . DIRECTORY_SEPARATOR . 'synthetic' . DIRECTORY_SEPARATOR . 'correctSamples';
+
+        $extractor = new TranslationExtractor($path);
+        $messages = $extractor->extract('defaultCategory');
+
+        $correctData = include 'data/correct-data.php';
+
+        $this->assertEquals($correctData, $messages);
+        $this->assertFalse($extractor->hasSkippedLines());
+    }
+
+    public function testExtractorWithOnlyIncorrectData(): void
+    {
+        $path = __DIR__ . DIRECTORY_SEPARATOR . 'extractorExamples' . DIRECTORY_SEPARATOR . 'synthetic' . DIRECTORY_SEPARATOR . 'incorrectSamples';
+
+        $extractor = new TranslationExtractor($path);
+
+        $messages = $extractor->extract();
+
+        $this->assertEquals([], $messages);
+        $this->assertTrue($extractor->hasSkippedLines());
+        $this->assertCount($this->incorrectDataCount, current($extractor->getSkippedLines()));
+    }
+
+    public function testExtractorWithOnlyBrokenData(): void
+    {
+        $path = __DIR__ . DIRECTORY_SEPARATOR . 'extractorExamples' . DIRECTORY_SEPARATOR . 'synthetic' . DIRECTORY_SEPARATOR . 'brokenSamples';
+
+        $extractor = new TranslationExtractor($path);
+
+        $messages = $extractor->extract();
+
+        $this->assertEquals(['' => ['messageId1']], $messages);
+        $this->assertTrue($extractor->hasSkippedLines());
+        $this->assertCount(1, current($extractor->getSkippedLines()));
+    }
+
+    public function testExtractorWithMixedData(): void
+    {
+        $path = __DIR__ . DIRECTORY_SEPARATOR . 'extractorExamples' . DIRECTORY_SEPARATOR . 'synthetic';
+
+        $extractor = new TranslationExtractor($path, null, ['**/brokenSamples/*']);
+
+        $messages = $extractor->extract('defaultCategory');
+
+        $correctData = include 'data/correct-data.php';
+
+        $this->assertEquals($correctData, $messages);
+        $this->assertTrue($extractor->hasSkippedLines());
+        $this->assertCount($this->incorrectDataCount, current($extractor->getSkippedLines()));
+    }
+
+    public function testExtractorWithMixedDataExcludeCorrect(): void
+    {
+        $path = __DIR__ . DIRECTORY_SEPARATOR . 'extractorExamples' . DIRECTORY_SEPARATOR . 'synthetic';
+
+        $extractor = new TranslationExtractor($path, ['**.php'], ['**/correctSamples/*', '**/brokenSamples/*']);
+
+        $messages = $extractor->extract('defaultCategory');
+
+        $this->assertEquals([], $messages);
+        $this->assertTrue($extractor->hasSkippedLines());
+        $this->assertCount($this->incorrectDataCount, current($extractor->getSkippedLines()));
+    }
+
+    public function testExtractorWithRealDataFromExtensionUser(): void
```

Comment:
```suggestion
    public function testExtractorWithRealDataFromUserExtension(): void
```

It is a good idea to link to the package repo.

Reply:
In real repository - count of messageId may be changed. Not good for tests